src/api/spec/jobs/status_history_rescaler_job_spec.rb

```diff
@@ -3,32 +3,52 @@
 RSpec.describe StatusHistoryRescalerJob, type: :job do
   include ActiveJob::TestHelper
 
+  let(:status_histories) { StatusHistory.where(key: 'busy_x86_64').order(:time) }
+
   describe '#rescale' do
-    before do
-      Timecop.freeze(2010, 7, 12)
-
-      now = Time.now.to_i - 2.days
-      StatusHistory.transaction do
-        1000.times do |i|
-          StatusHistory.create(time: now + i, key: 'idle_x86_64', value: i)
-        end
+    context 'newer than 2 days events' do
+      before do
+        5.times { |i| StatusHistory.create(time: (Time.now.utc - i.seconds).to_i, key: 'busy_x86_64', value: 10 * i) }
       end
 
-      StatusHistory.create(time: Time.now.to_i, key: 'busy_x86_64', value: 100)
+      subject! { StatusHistoryRescalerJob.perform_now }
+
+      it 'keeps the StatusHistory with key = busy_x86_64' do
+        expect(status_histories.count).to eq(5)
+        expect(status_histories.first.value).to eq(40)
+        expect(status_histories.last.value).to eq(0.0)
+      end
     end
 
-    after do
-      Timecop.return
+    context 'newer than 9 days' do
+      before do
+        2.times { |i| StatusHistory.create(time: 9.days.ago + i.minutes, key: 'busy_x86_64', value: 10 * i) }
```

Comment:
Here just 2 records are older than 7 days, so they get reduced to 1 (both are merged together).